solarforecastarbiter/reports/figures.py

```diff
@@ -411,6 +411,41 @@ def bar(cds, metric):
     return fig
 
 
+def calc_y_start_end(y_min, y_max, pad_factor=1.03):
+    """
+    Determine y axis start, end.
+
+    Parameters
+    ----------
+    y_min : float
+    y_max : float
+    pad_factor : float
+        Number by which to multiply the start, end.
+
+    Returns
+    -------
+    start, end : float, float
+    """
+    # bokeh does not play well with nans
+    y_min = np.nan_to_num(y_min)
```

Comment:
I think this should be moved into the min/max generator. Otherwise, consider when some metric has a nan in the data, but other data items have no nans. Then both y_min and y_max will be nan and the limits will be 0 to 0 when they shouldn't be. Right?

Reply:
Hm, I think we need it in both places. Now the generators are equivalent to lines like:

```python
In [73]: min(np.nanmin(np.array([np.nan, -1, 1])), np.nanmin(np.array([np.nan, -2, 2])))
Out[73]: -2.0

In [74]: min(np.nanmin(np.array([np.nan, -1, 1])), np.nanmin(np.array([np.nan, np.nan, np.nan])))
/Users/holmgren/miniconda3/envs/sfacore/bin/ipython:1: RuntimeWarning: All-NaN slice encountered
  #!/Users/holmgren/miniconda3/envs/sfacore/bin/python
Out[74]: -1.0

In [75]: min(np.nanmin(np.array([np.nan, np.nan, np.nan])), np.nanmin(np.array([np.nan, np.nan, np.nan])))
/Users/holmgren/miniconda3/envs/sfacore/bin/ipython:1: RuntimeWarning: All-NaN slice encountered
  #!/Users/holmgren/miniconda3/envs/sfacore/bin/python
Out[75]: nan
```

So this `np.nan_to_num(y_min)` handles the last case.